src/tactic/core.lean

```diff
@@ -64,14 +66,32 @@ end expr
 namespace interaction_monad
 open result
 
+variables {σ : Type} {α : Type u}
+
+/-- `get_state` returns the underlying state inside an interaction monad, from within that monad. -/
+meta def get_state : interaction_monad σ σ :=
```

Comment:
I know there's `tactic.read` and `tactic.write`. If it's valuable to generalise to `interaction_monad`, then presumably they should be combined or at least cross-referenced?

Reply:
Yes indeed! But they are in core, so are untouchable. The only question is whether to copy their names, and I think no for two reasons; 1) oftentimes one has `interaction_monad` and `tactic` both open, so this could cause a name resolution ambiguity(?) and 2) I think the names `read` and `write` are relatively poor names for what is being done, e.g. `io(.fs).read/write` reads/writes to a file instead! I would assert in general that, with respect to computers, `read`ing and `writ`ing comes usually with the connotation of accessing something external to the local context.

Reply:
Perhaps not "oftentimes", but it certainly happens.